OracleWebLogic/samples/12213-webtier-apache/Dockerfile

```diff
@@ -46,16 +47,41 @@ COPY container-scripts/* /u01/oracle/container-scripts/
 
 # Use unzip because the base image does not contain a JDK
 # Note that adding unzip does not noticeably increase the size of the image
-RUN yum install -y unzip httpd libaio mod_ssl.x86_64 && \
+RUN yum install -y sudo unzip httpd libaio mod_ssl.x86_64
+
+RUN groupadd sudo && \
+    useradd -b /u01 -d /u01/oracle -m -G sudo -s /bin/bash oracle && \
+    touch /etc/sudoers && \
+    sed -i /etc/sudoers -re 's/^%sudo.*/%sudo ALL=(ALL:ALL) NOPASSWD: ALL/g' && \
+    sed -i /etc/sudoers -re 's/^root.*/root ALL=(ALL:ALL) NOPASSWD: ALL/g' && \
+    sed -i /etc/sudoers -re 's/^#includedir.*/## **Removed the include directive** ##"/g' && \
+    echo "oracle ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
     unzip /tmp/$FMW_PKG -d /tmp && \
     unzip /tmp/$PLUGINS_PKG -d /tmp && \
     unzip /tmp/$PLUGIN_PKG -d $PLUGIN_HOME && \
     sed -i -e "s;logs/error_log;/proc/self/fd/2;" /etc/httpd/conf/httpd.conf && \
     sed -i -e "s;logs/access_log;/proc/self/fd/1;" /etc/httpd/conf/httpd.conf && \
-    rm /tmp/$PLUGINS_PKG /tmp/WLSPlugin12.2.1.3.0-*.zip /tmp/$FMW_PKG /tmp/fmw_12213_readme*
+    sed -i -e "s;User apache;User oracle;" /etc/httpd/conf/httpd.conf && \
+    sed -i -e "s;Group apache;Group oracle;" /etc/httpd/conf/httpd.conf && \
+    sed -i -e "s;Listen 80;Listen 8080;" /etc/httpd/conf/httpd.conf  && \
+    sed -i -e "s;root apache;oracle oracle;" /usr/lib/tmpfiles.d/httpd.conf && \
+    sed -i -e "s;apache apache;oracle oracle;" /usr/lib/tmpfiles.d/httpd.conf && \
+    chown -R oracle /var/log/httpd && \
+    chown oracle  -R /etc/httpd/logs/ && \
+    chown oracle -R /etc/pki/tls/certs/localhost.crt && \
+    chown oracle -R /etc/pki/tls/private/localhost.key && \
+    setcap cap_net_bind_service=+epi /usr/sbin/httpd && \
+    ldconfig && \
+    rm /tmp/$PLUGINS_PKG /tmp/WLSPlugin12.2.1.3.0-*.zip /tmp/$FMW_PKG /tmp/fmw_12213_readme* && \
+    chown oracle:oracle -R /u01 && \
+    chown oracle:oracle -R /tmp && \
+    chown oracle:oracle -R /config && \
+    chown oracle:oracle -R /run
 
 #Expose ports
-EXPOSE 80 4433
+EXPOSE 8080 4433
```

Comment:
Will this cause problems for the existing customers when they upgrade to the new version?

Reply:
When customers upgrade to next version, for eg., say 12.2.1.4, then we plan to provide support for both 80 and 8080.

Reply:
I would like to make this a generic image for 12.2.1.3, 12.2.14, 14.1.1.0  can we support both 80 and 8080 now?